### DISCUSSION TITLE
permissions: differentiate between "folder" and "file" write permissions

### DISCUSSION THREAD

**Discussion by mewalig** [Studio-42/elFinder#3348] [Q&A]
This is similar to #3211. I would like to designate certain folders such that:
- user can move a file into the folder
- user cannot create a subfolder in the folder

It seems to me that a simple change to enable this would be add an option to handle subfolder creation. For backwards-compatibility, a falsy value would not change the current behavior, and a non-falsy value would prevent subdirectory writes (or possibly provide further granularity).

As an example, if `readonly_subdir` is the new option, then commands/mkdir.js would be changed such that for any x, references to `x.write` become `(x.write && !x.readonly_subdir)`.

Can we create a pull request and get this change made?

**Comment by nao-pon** [comment]
@mewalig The connector can tell the client the options for that folder with the Options key for each folder.

You can disable the UI-side mkdir command in a particular folder by returning `$stat['options']['disabled'] = ['mkdir', ...]`.

**Reply by mewalig** [reply]
Will try it out, thank you
